docs/vm-configuration.md

```diff
@@ -264,3 +264,312 @@ the generation of multiple libvirt XML documents, the domain XML and associated
+    clock:
+      - timer: hypervclock
+        present: true
+      - timer: rtc
+        tickPolicy: catchup
+      - timer: pit
+        tickPolicy: delay
+      - timer: hpet
+        present: false
+    features:
+      - feature: acpi
+      - feature: hyperv
+        relaxed: true
+        vapic: true
+        spinlocks:
+          retries: 8191
+          state: true
+    cpu:
+      match: exact
+      model: Westmere
+      topology:
+        cores: 1
+        sockets: 16
+        threads: 1
+      numa:
+        - cell:
+          cpus: [1, 2, 3, 4]
+          memory: 4194304
+    devices:
+     channels:
+       - type: unix
+         source:
+           mode: bind
+           path: /var/lib/libvirt/qemu/channels/11111111-feed-dead-beef-111111111111.org.qemu.guest_agent.0
+         target:
+           name: org.qemu.guest_agent.0
+           type: virtio
+       - type: spicevmc
+         target:
+           name: com.redhat.spice.0
+           type: virtio
+     memballoon:
+       model: virtio
+     rng:
+       backend: random
+       path: /dev/urandom
+     controllers:
+       - type: scsi
+         model: virtio-scsi
+      video:
+      - type: qxl
+        ram: 65536
+        vgamem: 16384
+        vram: 8192
+      disks:
+      - device: cdrom
+        snapshot: false
+        type: file
+        source:
+          startupPolicy: optional
+        target:
+          bus: ide
+          dev: hdc
+        readonly: true
+      - device: disk 
+        snapshot: false
+        type: file
+        source:
+          file: /mntpoint/11111111-1111-1111-1111-111111111111/images/22222222-2222-2222-2222-222222222222/22222222-2222-2222-2222-222222222222
+        target:
+          bus: scsi
+          dev: sda
+        bootOrder: 1
+        driver:
+          cache: none
+          name: qemu
+          type: raw
+          io: threads
+      interfaces:
+      - source:
+          bridge: mgmtnet
+        type: bridge
+        model: virtio
+        mac:
+          address: 00:aa:bb:cc:dd:ee
+        filterRef:
+          filter: no-mac-spoofing
+      graphics:
+      - listen:
+          network: mgmnet
+          type: network
+        port: -1
+        type: spice
+        tlsPort: -1
+        passwd: *****
+        passwdValidTo: 1970-01-01T00:00:01
+    os:
+      type:
+        os: hvm
+        arch: x86_64
+        machine: pc-i440fx-rhel7.3.0
+      smbios:
+        mode: sysinfo
+```
+
+This VM Specification requires users (where users include management
+applications) to know and re-implement part of libvirt and OS level logic.
+Instead of asking users to do so, we could simplify the spec generation by
+adding a layer of high level abstraction.
+
+This is an example of features section describing the VM Specification above:
+
+```yaml
+apiVersion: kubevirt.io/v1alpha1
+kind: VM
+metadata:
+  name: VM
+spec:
+  features:
+    memory: 4096 # other memory attributes can be deducted
```

Comment:
This is an oversimplification. You will definitely need current, minimal and maximal memory size. And we should probably use the same quantity format k8s uses (string with units eg. "10G")

Reply:
sure, it's just an example. It should also include hugepages backing configuration, min mem guaranteed